app/main/controllers/projects.py

```diff
@@ -158,54 +183,105 @@ def get_project_sources(user_id, project_id):
             })
 
         pattern = '%' + search_query + '%'
-        results = Source.query.filter(Source.project_id == project_id)\
+        results = Source.query.filter(Source.project_id == project_id) \
             .filter(Source.url.ilike(pattern)
                     | Source.title.ilike(pattern)
                     | Source.content.ilike(pattern)
-                    | Source.highlights.ilike(pattern)
-                    | Source.notes.ilike(pattern)).order_by(Source.id).all()
+                    | Source.child_items.ilike(pattern)).order_by(Source.id).all()
 
         return jsonify({
             'success': True,
             'sources': [source.format_short() for source in results]
         })
 
+
+
+    """
+    Gets all the items of a given project
+    or searches through them if a search query is passed.
+    Search looks at all text fields of a source.
     """
-    Creates a new source inside an existing project.
+    @app.route('/projects/<int:project_id>/items')
+    @requires_auth('read:items')
+    def get_project_items(user_id, project_id):
+        project = get_authorized_project(user_id, project_id)
+
+        search_query = request.args.get('query', None)
+        if search_query is None:
+            # Returns all items
+            return jsonify({
+                'success': True,
+                'sources': [i.format() for i in project.items]
+            })
+
+        pattern = '%' + search_query + '%'
+        results = Item.query.filter(Item.parent_project == project_id) \
+            .filter(Item.source.url.ilike(pattern)
+                    | Item.source.title.ilike(pattern)
+                    | Item.source.content.ilike(pattern)
+                    | Item.content.ilike(pattern)).order_by(Item.source.id).all()
+
+        return jsonify({
+            'success': True,
+            'items': [item.format() for item in results]
+        })
+
+    """
+    Creates a new item inside an existing project.
     The necessary data is passed as a JSON body.
     """
-    @app.route('/projects/<int:project_id>/sources', methods=['POST'])
-    @requires_auth('create:sources')
-    def create_source(user_id, project_id):
-        get_authorized_project(user_id, project_id)
+
+    @app.route('/items', methods=['POST'])
+    @requires_auth('create:items')
+    def create_items(user_id):
 
         body = request.get_json()
         if body is None:
             abort(400)
 
         url = body.get('url', None)
+        content = body.get('content', None)
+        project_id = body.get('parent_project', None)
         x = body.get('x_position', None)
         y = body.get('y_position', None)
-        if url is None:
+        get_authorized_project(user_id, project_id)
```

Comment:
Are you doing anything with the output of this function?

Reply:
For now, it simply checks to see if the user is authorized to add to the project.  This may have to be changed (because of your other comment about clusters)